crud_handler/crud_handler.go

```diff
@@ -0,0 +1,191 @@
+package crud_handler
+
+import (
+	"encoding/json"
+	"fmt"
+	"main/transformer"
+	"net/http"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/google/uuid"
+)
+
+type TransformRequest struct {
+	Type        string `json:"type"`
+	CaesarShift int    `json:"shift,omitempty"`
+	Input       string `json:"input,omitempty"`
+}
+
+type TransformResult struct {
+	Id          string `json:"uuid"`
+	Type        string `json:"type"`
+	CaesarShift int    `json:"shift,omitempty"`
+	Result      string `json:"result,omitempty"`
+	Created_at  int64  `json:"created"`
+	Updated_at  int64  `json:"updated,omitempty"`
+}
+
+// Fake database just for example
+var temp1, temp2, temp3 = uuid.NewString(), uuid.NewString(), uuid.NewString()
+var TransformRequests = map[string]*TransformRequest{
+	temp1: &TransformRequest{Type: "reverse", Input: "123456789"},
+	temp2: &TransformRequest{Type: "base64", Input: "Man"},
+	temp3: &TransformRequest{Type: "caesar", CaesarShift: 1, Input: "zab"},
+}
+var TransformResults = map[string]*TransformResult{
+	temp1: &TransformResult{Id: temp1, Type: "reverse", Result: "987654321", Created_at: time.Now().Unix()},
+	temp2: &TransformResult{Id: temp2, Type: "base64", Result: "TWFu", Created_at: time.Now().Unix() + 1},
+	temp3: &TransformResult{Id: temp3, Type: "caesar", CaesarShift: 1, Result: "abc", Created_at: time.Now().Unix() + 2},
+}
+
+func SetJSONContentType(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
setting response content-type from middleware is not obvious, create helper for is and us it in handler instead

Reply:
still actual